docs/guide/search/extend_elasticsearch.md

```diff
@@ -0,0 +1,422 @@
+
+!!! enterprise
+
+    ## Indexing custom data
+
+    [Elasticsearch](elastic.md) indexes content and Location data out of the box.
+    Besides what is indexed automatically, you can add additional data to the Elasticsearch index.
+
+    To do so, subscribe to one of the following events:
+
+    - `Ibexa\Platform\Contracts\ElasticSearchEngine\Mapping\Event\ContentIndexCreateEvent`
+    - `Ibexa\Platform\Contracts\ElasticSearchEngine\Mapping\Event\LocationIndexCreateEvent`
+
+    These events are called when the index is created for the content and Location documents, respectively.
+
+    You can pass the event to a subscriber which gives you access to the document that you can modify.
+
+    In the following example when an index in created for a content or a Location document,
+    the event subscriber adds a `custom_field` of the type `StringField` to the index:
+
+    ``` php hl_lines="19 20 21"
+    <?php
+
+    declare(strict_types=1);
+
+    namespace App\EventSubscriber;
+
+    use eZ\Publish\SPI\Search\Field;
+    use eZ\Publish\SPI\Search\FieldType\StringField;
+    use Ibexa\Platform\Contracts\ElasticSearchEngine\Mapping\Event\ContentIndexCreateEvent;
+    use Ibexa\Platform\Contracts\ElasticSearchEngine\Mapping\Event\LocationIndexCreateEvent;
+    use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+    final class CustomIndexDataSubscriber implements EventSubscriberInterface
+    {
+        public function onContentDocumentCreate(ContentIndexCreateEvent $event): void
+        {
+            $document = $event->getDocument();
+            $document->fields[] = new Field(
+                'custom_field',
+                'Custom field value',
+                new StringField()
+            );
+        }
+
+        public function onLocationDocumentCreate(LocationIndexCreateEvent $event): void
+        {
+            $document = $event->getDocument();
+            $document->fields[] = new Field(
+                'custom_field',
+                'Custom field value',
+                new StringField()
+            );
+        }
+
+        public static function getSubscribedEvents(): array
+        {
+            return [
+                ContentIndexCreateEvent::class => 'onContentDocumentCreate',
+                LocationIndexCreateEvent::class => 'onLocationDocumentCreate'
+            ];
+        }
+    }
+    ```
+
+    Remember to register the subscriber as a service:
+
+    ``` yaml
+    services:
+        App\EventSubscriber\CustomIndexDataSubscriber:
+            tags:
+                - { name: kernel.event_subscriber }
+    ```
+
+    ## Manipulating the query
+
+    You can customize the search query before it is executed.
+    To do it, subscribe to `Ibexa\Platform\Contracts\ElasticSearchEngine\Query\Event\QueryFilterEvent`.
+
+    The following example shows how to add an additional Search Criterion to all queries.
+
+    Note that depending on your configuration, this will impact all search queries,
```

Comment:
```suggestion
    Note that depending on your configuration, this might impact all search queries,
```

Does the statement call for a note? If so, I'd add a Note section. 
Otherwise you could put the two sentences in one paragraph, and remove the "Note that..." to make it flow better and lessen the ambiguity of "this"